inference_perf/loadgen/load_generator.py

```diff
@@ -177,9 +205,24 @@ def _sigint_handler(self, _signum: int, _frame: Optional[FrameType]) -> None:
         """SIGINT handler that sets interrup_sig flag to True"""
         self.interrupt_sig = True
 
+    def _set_worker_concurrency(self, concurrency_level: int) -> None:
+        # Calculate new concurrency for worker (concurrency_level will always be > 0)
+        new_concurrency = concurrency_level // self.num_workers + 1
+        # Calculate index cutoff for workers with +1 concurrency
+        worker_less_load_id = (
+            concurrency_level % self.num_workers if concurrency_level % self.num_workers != 0 else float("inf")
+        )
+        for worker in self.workers:
+            worker_concurrency = new_concurrency if worker.id < worker_less_load_id else new_concurrency - 1
+            worker.max_concurrency = worker_concurrency
```

Comment:
I think we can replace each worker.max_concurrency with the Synchronized[int], loadgen shouldn't need both paths. The non-concurrency load will just only set the max_concurrency once.